platform/root/__tests__/index.test.ts

```diff
@@ -98,3 +98,30 @@ test('calls onShutdown param on "shutdown"', async () => {
   expect(onShutdown).toHaveBeenCalledTimes(1);
   expect(onShutdown).toHaveBeenLastCalledWith(err);
 });
+
+test('calls shutdown when configuring logger fails', async () => {
+  const logged = jest.spyOn(console, 'error');
+  logged.mockImplementation(noop);
+
+  const onShutdown = jest.fn();
+
+  const root = new Root(config$, env, onShutdown);
+  const err = new Error('foo bar baz');
+
+  configService.atPath.mockImplementationOnce(() => {
+    throw err;
+  });
+
+  mockServer.mockClear();
+
+  await root.start();
+
+  expect(mockServer).not.toHaveBeenCalled();
+
+  expect(onShutdown).toHaveBeenCalledTimes(1);
+  expect(onShutdown).toHaveBeenLastCalledWith(err);
+
+  expect(logged.mock.calls).toMatchSnapshot();
+
+  logged.mockRestore();
```

Comment:
__issue:__ the problem with restoring globals inside the test is that when test fails spy won't be restored and the following tests will use unrestored spy. So usually one should create spies in the "parent" scope or inside `beforeEach` and restore in them `afterEach` (same pattern can be applied to the `process.exit` above btw).